### ISSUE TITLE
Duplicate `__typename` in response for nested union queries

### ISSUE THREAD

**Issue by dkempner** [99designs/gqlgen#3374]
### What happened?

When running a query with this shape, the response that's returned is _technically_ not valid JSON because it has duplicate `__typename` properties.

#### Query

```
query {
  union {
    __typename
    ... on A {
      __typename
    }
    ... on B {
      __typename
    }
  }
}
```

#### Response

```
{
    "data": {
        "union": [
            {
                "__typename": "A",
                "__typename": "A"
            },
            {
                "__typename": "B",
                "__typename": "B"
            }
        ]
    }
}
```

### What did you expect?

Duplicate properties should be deduplicated.

### Minimal graphql.schema and models to reproduce
1. Clone https://github.com/dkempner/gqlgen-union-double-typename
2. `go run server.go`
3. Visit http://localhost:8080/
4. Open Chrome devtools to network tab
5. Execute the query saved in [repro.grapqhl](https://github.com/dkempner/gqlgen-union-double-typename/blob/master/repro.graphql)
6. Observe the "Response" tab has duplicate `__typename` properties

<img width="2536" alt="image" src="https://github.com/user-attachments/assets/b600d749-9604-43f6-b2bc-86e8969cd5b9">

### versions
 - `go run github.com/99designs/gqlgen version`? 
   - `v0.17.56`
 - `go version`? 
   - `go version go1.23.3 darwin/arm64`

**Comment by StevenACoffman** [comment]
Oh, wow. Yeah, that's definitely a bug. Thanks for reporting, and PR wanted.

**Comment by ActuallyConnor** [comment]
It looks like this is no longer an issue. Should the issue be closed?

**Comment by ActuallyConnor** [comment]
I downloaded the example to try and reproduce and did not find the issue

**Comment by dkempner** [comment]
@ActuallyConnor, I updated to the latest version of `gqlgen` (`v0.17.70`) and still had repro. 

Are you sure you followed the repro instructions? The "Preview" tab in Chrome Devtools will show the expected output, but the "Response" tab will not.

# Preview Tab

<img width="558" alt="Image" src="https://github.com/user-attachments/assets/5b35e073-4295-475a-a178-38ce8eea6e39" />

# Response Tab

<img width="564" alt="Image" src="https://github.com/user-attachments/assets/cf61a890-58ef-4aad-9bcb-204b3774de7f" />

**Comment by ActuallyConnor** [comment]
@dkempner is your repo updated too?

**Comment by dkempner** [comment]
> [@dkempner](https://github.com/dkempner) is your repo updated too? I can try to debug if it is.

Yep @ActuallyConnor the [repository](https://github.com/dkempner/gqlgen-union-double-typename) is updated to latest version!

**Comment by StevenACoffman** [comment]
Ok, I would love to have a PR if either of you are able to figure out the issue.

**Comment by StevenACoffman** [comment]
@giautm Yeah, you are entirely correct! A PR would be very welcome, especially since I am currently sick and feverish and don't trust myself to do it correctly right now.

**Comment by mattcarranza10** [comment]
@StevenACoffman A review is welcome.

**Comment by mattcarranza10** [comment]
@StevenACoffman Just a quickly reminder, this issue can be closed.
